service/history/handler.go

```diff
@@ -550,6 +551,14 @@ func (h *Handler) RespondDecisionTaskFailed(
 		token.RunID,
 		token.ScheduleID))
 
+	if failedRequest != nil && failedRequest.GetCause() == workflow.DecisionTaskFailedCauseUnhandledDecision {
+		h.GetLogger().Info("Non-Deterministic Error", tag.WorkflowRunID(token.RunID))
+		domainName, err := h.GetDomainCache().GetDomainName(token.DomainID)
+		if err == nil {
+			nonDeterministic := h.GetMetricsClient().Scope(metrics.HistoryRespondDecisionTaskFailedScope, metrics.DomainTag(domainName))
+			nonDeterministic.IncCounter(metrics.NonDeterministicErrorCount)
```

Comment:
`scope` is already defined on L527, you can reuse its value here. Also it's not necessary to create a temp variable `nonDeterministic`.